### PR TITLE
Add feature: only download image via Wi-Fi

### LINKED ISSUE #1502: Feature request: Enable image download on WiFi
### Expected behavior
Provide image download 3 options
Always, only WiFi, 

### Environment
K-9 Mail version:5.010

Android version:

Account type (IMAP, POP3, WebDAV/Exchange):

### PR DESCRIPTION
**Pull request by daquexian**
Fixes #1502 

Add a feature that only download images via Wi-Fi

Since this option should be parallel with "Always show images" (the option "only download image via Wi-Fi" should have its own value regardless of the value of "Always show images"), so a separate checkbox is added.

### CODE REVIEW

**Comment by philipwhiuk** [comment]
Why not just another option to the ShowPictures enum?

**Comment by daquexian** [comment]
@philipwhiuk Thanks for your review :) I think that "download images from anyone only via Wi-Fi" and "download images from contacts only via Wi-Fi" are both possible to be needed by users. If just add it in `ShowPictures`, users will not be able to set whether only the images from contacts should be downloaded.

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java [hunk 1]
```diff
@@ -300,6 +306,19 @@ private boolean shouldShowPicturesFromSender(ShowPictures showPicturesSetting, M
...
+        if (!onlyShowViaWiFi) {
+            return true;
+        }
+
+        final ConnectivityManager connectivityManager =
+                (ConnectivityManager) getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkType networkType =
+                NetworkType.fromConnectivityManagerType(connectivityManager.getActiveNetworkInfo().getType());
+
+        return networkType == NetworkType.WIFI;
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java]: Despite the name, this should probably be [isActiveNetworkMetered](https://developer.android.com/reference/android/net/ConnectivityManager.html#isActiveNetworkMetered), which is what 99% of people mean when they say "only on wifi". @cketti ?

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java [hunk 2]
```diff
@@ -282,8 +285,11 @@ private void hideShowPicturesButton() {
         showPicturesButton.setVisibility(View.GONE);
     }
 
-    private boolean shouldAutomaticallyLoadPictures(ShowPictures showPicturesSetting, Message message) {
-        return showPicturesSetting == ShowPictures.ALWAYS || shouldShowPicturesFromSender(showPicturesSetting, message);
+    private boolean shouldAutomaticallyLoadPictures(ShowPictures showPicturesSetting, boolean onlyShowPicturesViaWiFi,
+            Message message) {
+        return (showPicturesSetting == ShowPictures.ALWAYS ||
+                shouldShowPicturesFromSender(showPicturesSetting, message)) &&
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java]: is this really an `&&`?

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java (hunk 1 above)
Comment (daquexian) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java]: Good idea :)

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java (hunk 2 above)
Comment (daquexian) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java]: I think so.. 
```
showPicturesSetting == ShowPictures.ALWAYS || shouldShowPicturesFromSender(showPicturesSetting, message)
```
indicates whether the user want images from this sender to show, 
```
shouldShowPicturesInCurrentNetwork(onlyShowPicturesViaWiFi)
```
indicates whether the user want images to show on current network. Images should be shown when both are met.

**Comment by daquexian** [comment]
@Valodim about the separate checkbox, my opinion is that we will not be able to meet the request like "only download images only when images are from contacts and I'm on Wi-Fi" if we just add an extra option in "show pictures never / always".

**Comment by Valodim** [comment]
I see. Hmm that's a problem, so we have all of these as separate option bits?
- only for contact
- only on wifi
- always

hmm, that's tricky, seems like there will always be constellations that are confusing to the user. Maybe we can solve this via wording, e.g. "Download automatically only while on wifi"?

**Comment by daquexian** [comment]
@Valodim I have made requested changes about code. As for the settings, in my opinion users will also confused about "what will happen if I check both 'Download automatically only while on wifi' and 'Download automatically only from contacts'?". But I have no better idea. :P So I will update this PR as soon as possible if you have determined the way to present these settings.

**Comment by daquexian** [comment]
@Valodim About the settings, do you have a final decision? :)

**Comment by cketti** [comment]
The approach looks good to me. Unfortunately, our code has changed quite a bit and it's probably easier to reimplement the feature instead of trying to rebase this pull request.

**Pull request closed without merging**